src/steinertree/steiner_tree.jl

```diff
@@ -0,0 +1,84 @@
+"""
+    filter_non_term_leaves!(g, term_vert)
+
+Remove edges of `g` so that all non-isolated leaves of `g` are in the set `term_vert`
+"""
+function filter_non_term_leaves!(
+    g::AbstractGraph{T},
+    term_vert::Vector{<:Integer}
+    ) where T<:Integer
+
+    is_term = falses(nv(g))
+    is_term[term_vert] .= true
+    leaves = [v for v in vertices(g) if degree(g, v) == 1 && !is_term[v]]
+
+    while !isempty(leaves)
+        leaf = pop!(leaves)
+        for v in neighbors(g, leaf)
+            rem_edge!(g, v, leaf)
+            if degree(g, v) == 1 && !is_term[v]
+                push!(leaves, v)
+            end
+        end
+    end
+
+    return g
+end
+
+"""
+	steiner_tree(g, term_vert, distmx=weights(g))
+
+Return a minimum weight spanning graph of connected, undirected graph `g` with positive edge 
+weights represented by `distmx` using [Approximate Steiner Tree](https://en.wikipedia.org/wiki/Steiner_tree_problem#Approximating_the_Steiner_tree).
+`t = length(term_vert)`
+
+### Performance
+Runtime: O(t*(t*log(t)+|E|*log(|V| ))
+Memory: O(t*|V|)
+Approximation Factor: 2-2/t
+"""
+function steiner_tree end
+
+@traitfn function steiner_tree(
```

Comment:
Wouldn't it be better to call this `steiner_tree_approximation` or `approximate_steiner_tree`?  Otherwise one would think, that this solved the problem exactly.

Reply:
That would have to be done for all algorithms to do not return the optimal answer. There, we use structs to call a particular algorithm. Eg. `vertex_cover(g, DegreeVertexCover())`.
There doesn't seem much point in doing this since only one algorithm is available.

Also, the name suggests a steiner tree is returned, but no mention of the minimum steiner tree.

I could do something like 
`steiner_tree(g, term_vert, Approx(), distmx=weights(g))`

Reply:
I think there was some talk about a unified problem interface recently, and in the meantime we just use your name, or my suggestions.

Reply:
I'll keep the API like this for now.

I changed the doc to `Return an approximately minimum steiner tree`